client/app/scripts/components/terminal.js

```diff
@@ -29,24 +27,35 @@ function ab2str(buf) {
 }
 
 function terminalCellSize(wrapperNode, rows, cols) {
-  //
-  // wrapperNode should be an unsized block. E.g. An element that 'clings' to
-  // its inner terminal so we get a nice size reading of the inner terminal.
-  //
-  const width = wrapperNode.clientWidth;
   const height = wrapperNode.clientHeight;
-  const pixelPerRow = height / rows;
+
+  // Guess the width of the row.
+  let width = wrapperNode.clientWidth;
+  // Now try and measure the first row we find.
+  const firstRow = wrapperNode.querySelector('.terminal div');
+  if (!firstRow) {
+    log("ERROR: Couldn't find first row, resizing might not work very well.");
+  } else {
+    const rowDisplay = firstRow.style.display;
+    firstRow.style.display = 'inline';
+    width = firstRow.offsetWidth;
```

Comment:
wrapperNode is a block div which fills its parent and gives a bad size reading.

We wanna get the container for the the actual text characters, turn it into a span so that it "clings" to them, and measure that. I found this clever hack on term.js's issue page somewhere.

Which makes me thing actually, that maybe if the initial 80 cols don't fit in the initial container they might wrap, giving another bad size reading. Maybe we should set the initial cols to 1... it will updated immediately to fit the window size anywho.. hmm